### PR TITLE
Do not ignore x2's title

### PR DESCRIPTION
**Pull request by kanitw**
Fix https://github.com/vega/vega-lite/issues/3742

### CODE REVIEW

File: test/compile/axis/parse.test.ts [hunk 1]
```diff
@@ -146,6 +164,27 @@ describe('Axis', function() {
       assert.equal(axisComponent['x'][0].explicit.title, 'foo');
     });
 
+    it('should merge title of fieldDef and fieldDef2', function () {
+      const model = parseUnitModelWithScale({
+        mark: "bar",
+        encoding: {
+          x: {
```
[the remark this answers is not in the document]
Comment (kanitw) [reply test/compile/axis/parse.test.ts]: That doesn't work for other reasons, and I'm not gonna fix it in this PR.  Feel free to file another issue.

File: test/compile/axis/parse.test.ts [hunk 2]
```diff
@@ -130,6 +130,24 @@ describe('Axis', function() {
       }
     });
 
+    it('should store the fieldDef title value if title = null, "", or false', function () {
```
[the remark this answers is not in the document]
Comment (kanitw) [reply test/compile/axis/parse.test.ts]: You gonna call another clean all service later anyway, so I'll ignore this :p

[the remark this answers is not in the document]
Comment (domoritz) [reply test/compile/axis/parse.test.ts]: Okay

File: test/compile/axis/parse.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (domoritz) [reply test/compile/axis/parse.test.ts]: Okay

File: src/compile/axis/parse.ts [hunk 3]
```diff
@@ -157,18 +157,21 @@ function getFieldDefTitle(model: UnitModel, channel: 'x' | 'y') {
...
+  const title1 = fieldDef ? fieldDef.title : undefined;
+  const title2 = fieldDef2 ? fieldDef2.title : undefined;
+
+  if (title1 && title2) {
+    return mergeTitle(title1, title2);
+  } else if (title1) {
+    return title1;
+  } else if (title2) {
+    return title2;
+  } else if (title1 !== undefined) {
```
Comment (domoritz) [inline src/compile/axis/parse.ts]: What is this about? Can you add a comment?

**Pull request merged**